src/app/services/shell.service.ts

```diff
@@ -19,29 +27,52 @@ export class ShellService {
     command: string;
   };
 
-  private shellCmdOutput: ArrayBuffer;
   shellOutput = new EventEmitter<ArrayBuffer>();
   shellConnected = new EventEmitter<ShellConnectedEvent>();
```

Comment:
`shellConnected` and `shellOutput` should be private and exposed with getters that only return the underlying observable with `asObservable()` instead of the subject itself as the `next` method can be called on these `EventEmitter` objects which exposes parts of the API we don't want exposed.